scaffolding-ruby/doc/reference.md

```diff
@@ -0,0 +1,473 @@
+# Ruby Scaffolding
+
+Ruby Scaffolding is a [Habitat][habitat] package which helps you build your Ruby-based **web applications**, **services**, and **processes** (hereafter referred to as "apps") into a package which runs consistently on a wide range of containers, virtual machines, or servers via the Habitat Supervisor. The Supervisor will facilitate clustering, discovery of database services, dynamically update configuration and credentials, coordinate reliable rolling updates, and a lot more!
+
+For more about Habitat, you can check out [Try Habitat][try_hab]. For more about building your apps, read on!
+
+## Use Ruby Scaffolding for Your App
+
+Check out the [Ruby Scaffolding QuickStart Guide](quickstart.md).
+
+## Scaffolding Detection
+
+To properly load and use this Scaffolding, the implementation looks for a [`Gemfile`][] in your app's root directory. Either directory structure is supported:
+
+A Plan in a `habitat/` subdirectory:
+
+```
+.
+├── Gemfile
+└── habitat
+    └── plan.sh
+```
+
+A Plan in the same directory as the `Gemfile`:
+
+```
+.
+├── Gemfile
+└── plan.sh
+```
+
+## Gemfile
+
+A [`Gemfile`][] must be present for this Scaffolding to function correctly. If no `Gemfile.lock` is present, a lock file will be generated by calling `bundle lock`. While committing a `Gemfile.lock` to source control is [highly recommended][commit_lockfile], it is not strictly required.
+
+## Package Dependencies
+
+Most non-trivial apps need more than their own codebase to run correctly. Many have gem dependencies which require compiling native gem extensions. Others require certain software present for shell out to (for example: ImageMagick). This Scaffolding uses app detection logic to conditionally inject some common Habitat packages into your Plan. However, if more packages are required, it is generally easy to add these to your Plan to fully describe your app's [build and runtime dependencies][12factor_dependencies].
+
+### Default Dependencies
+
+The following Habitat package dependencies will be injected into your app's Plan:
+
+* [`core/busybox-static`][]: Used by process bins to have valid [shebangs][] and a consistent minimal command set. Will be injected into your Plan's `pkg_deps` array.
```

Comment:
We can get away with using the really tiny, zero-dependency busybox-static package in this case because we are writing out the hooks. Additionally, since the Supervisor under Linux pulls this in, there's a really good chance that we don't install any more software. In this narrow case, we don't need the full Bash power.